### PR TITLE
Start setting up oauth flow

### PR DESCRIPTION
**Pull request by mattprecious**
TODO:
- [ ] Add a sign in button somewhere
- [ ] Make an authenticated request (account information for nav drawer header?)
- [ ] Handle 401s
- [ ] Sign out

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -118,7 +118,7 @@ dependencies {
   compile 'com.squareup.dagger:dagger:1.2.2'
   apt 'com.squareup.dagger:dagger-compiler:1.2.2'
 
-  compile 'com.squareup.okhttp:okhttp:2.3.0'
+  compile 'com.squareup.okhttp:okhttp:2.4.0'
```
Comment (JakeWharton) [inline build.gradle]: Hah... #192

File: src/internalDebug/java/com/jakewharton/u2020/data/DebugDataModule.java [hunk 2]
```diff
@@ -60,6 +61,12 @@ RxSharedPreferences provideRxSharedPreferences(SharedPreferences preferences) {
     return client;
   }
 
+  @Provides @Singleton @AccessToken StringPreference provideAccessToken(SharedPreferences prefs,
+      @ApiEndpoint StringPreference endpoint) {
+    // Return an endpoint-specific preference.
+    return new StringPreference(prefs, "access-token-" + endpoint.get());
```
Comment (JakeWharton) [inline src/internalDebug/java/com/jakewharton/u2020/data/DebugDataModule.java]: clever

File: src/main/java/com/jakewharton/u2020/ui/MainActivity.java [hunk 3]
```diff
@@ -93,6 +98,19 @@
...
   }
 
+  @Override protected void onNewIntent(Intent intent) {
+    super.onNewIntent(intent);
+
+    Uri data = intent.getData();
+    if (data == null) return;
+
+    if ("u2020".equals(data.getScheme())) {
+      Intent serviceIntent = new Intent(this, OauthService.class);
```
Comment (JakeWharton) [inline src/main/java/com/jakewharton/u2020/ui/MainActivity.java]: We might have talked about this, but can the service just register for the intent directly?

File: src/main/java/com/jakewharton/u2020/U2020Module.java [hunk 4]
```diff
@@ -13,7 +14,8 @@
         DataModule.class
     },
     injects = {
-        U2020App.class
+        U2020App.class,
+        OauthService.class
```
Comment (JakeWharton) [inline src/main/java/com/jakewharton/u2020/U2020Module.java]: This should probably be in `ApiModule` since that's the nearest module.

**Comment by JakeWharton** [comment]
lgtm, whether you want to merge now or keep adding to this PR

File: src/main/java/com/jakewharton/u2020/ui/MainActivity.java (hunk 3 above)
Comment (mattprecious) [inline src/main/java/com/jakewharton/u2020/ui/MainActivity.java]: We want U2020 to come to the front when we intercept the intent, so it makes sense to handle it with the activity.

(Not sure if this is possible, anyway. Couldn't get it to work with my quick attempt).

**Comment by mattprecious** [comment]
*merge on success*

**Pull request merged**